### PR TITLE
v2.0 docs update

### PR DESCRIPTION
**Pull request by dogversioning**
This PR adds documentation around some V2 features, and a generator for creating markdown tables.

### Checklist
- [X] Consider if documentation (like in `docs/`) needs to be updated
- [X] Consider if tests should be added
- [X] Update template repo if there are changes to study configuration

### CODE REVIEW

File: cumulus_library/studies/core/core_templates/observation.sql.jinja [hunk 1]
```diff
@@ -52,15 +51,12 @@ SELECT
     id,
     category_code,
     category_code_system,
-    component,
```
Comment (dogversioning) [inline cumulus_library/studies/core/core_templates/observation.sql.jinja:55]: the fields removed here were not flattened - i'll swing back around on the required/must support fields on these.

File: cumulus_library/cli.py [hunk 2]
```diff
@@ -373,6 +389,10 @@ def run_cli(args: dict):
             elif args["action"] == "generate-sql":
                 for target in args["target"]:
                     runner.generate_study_sql(study_dict[target])
+
+            elif args["action"] == "generate-md":
```
Comment (mikix) [inline cumulus_library/cli.py:393]: Ah - that's a nice analogy to `generate-sql` -- good job us on naming that one.

File: cumulus_library/template_sql/column_datatype.sql.jinja [hunk 3]
```diff
@@ -1,8 +1,13 @@
 SELECT
+    {%- if include_table_names %}
+    table_name,
+    {%- endif %}
```
Comment (mikix) [inline cumulus_library/template_sql/column_datatype.sql.jinja:4]: Could you unconditionally include it, if you did it last? (i.e. to not break any `[0]` references)

File: docs/creating-sql-with-python.md [hunk 4]
```diff
@@ -85,7 +94,10 @@ templates - instead, using the
 [template function library](https://github.com/smart-on-fhir/cumulus-library/blob/main/cumulus_library/template_sql/base_templates.py)
 you can provide arguments to these templates that will allow you to
 generate standard types of SQL tables, as well as using templates targeted for
-bespoke operations. 
+bespoke operations. But you :can: write study specific templates if you have
```
Comment (mikix) [inline docs/creating-sql-with-python.md:97]: nit: use `_can_` instead - we have actual typography here, we don't need hints-for-humans :smile:

File: docs/creating-sql-with-python.md [hunk 5]
```diff
@@ -85,7 +94,10 @@ templates - instead, using the
...
+a complex use case. The Core study has 
+[study specific templates](https://github.com/smart-on-fhir/cumulus-library)
```
Comment (mikix) [inline docs/creating-sql-with-python.md:99]: Is that the right URL?

File: docs/creating-studies.md [hunk 6]
```diff
@@ -24,64 +24,91 @@ to any build/export call to tell it where to look for your work.
...
+format, as we'll show you in this document how the library uses these files to run your
+study. You can copy the following template as an example, which has comments describing
+what each section does:
+
+```toml
+# 'study_prefix' should be a string at the start of each table. We'll use this
+# to clean up queries, so it should be unique. Name your tables in the following
+# format: [study_prefix]__[table_name]. It should probably, but not necessarily,
+# be the same name as the folder the study definition is in.
+>>>>>>> 1e357f9 (v2.0 docs update)
```
Comment (mikix) [inline docs/creating-studies.md:44]: :scissors:

File: docs/creating-studies.md [hunk 7]
```diff
@@ -105,12 +135,13 @@ Most users have a workflow that looks like this:
 #### sqlfluff
 
 We use [sqlfluff](https://github.com/sqlfluff/sqlfluff) to help maintain a consistent
-style across many different SQL query authors. We recommend using `sqlfluff` as you
-are developing your queries to ensure your SQL is matching the style of other
-authors. We copy over our `sqlfluff` rules when you use `cumulus-library` to create
-a study, so no additional configuration should be needed.
+style across many different SQL query authors. We recommend using sqlfluff as you
+are developing your queries to ensure your sql is matching the style of other
```
Comment (mikix) [inline docs/creating-studies.md:139]: nit: capitalize SQL

File: cumulus_library/study_parser.py [hunk 8]
```diff
@@ -552,6 +555,57 @@ def run_generate_sql(
...
+        )
+
+        tables = [x[0] for x in cursor.execute(query).fetchall()]
+        query = base_templates.get_column_datatype_query(
+            schema_name=schema, table_names=tables, include_table_names=True
+        )
+        study_df = pandas.DataFrame(
+            cursor.execute(query).fetchall(), columns=["Table", "Column", "Type"]
+        )
+        print(self._study_path)
```
Comment (mikix) [inline cumulus_library/study_parser.py:583]: :scissors:

File: docs/core-study-details.md [hunk 9]
```diff
@@ -8,98 +8,501 @@ nav_order: 5
...
+- **Flattened FHIR resources** - The core study provided flattened tables from nested FHIR resources, making queries easier to construct without having to worry about the nuances of conditional unnesting against potentially missing data
+- **Summary Exports** - The core study will provide some basic count data across the entire cohort that you've processed via ETL, which can be useful as a verification step for validating the integrity of the data you're extracting from your EHR system
 
-    #count (total) =
-    #count patients age 9 =
-    #count patients age 9 and rtPCR POS =
-    #count patients age 9 and rtPCR NEG =
-    #count rtPCR POS =
-    #count rtPCR NEG =
+If you are authoring a study, and are focused only on clinical analysis (i.e. you aren't working on data quality/data governance issues), we :strongly: recommend you use the core study as the starting point for your own work. See
```
Comment (mikix) [inline docs/core-study-details.md:17]: nit: treat yourself to some bold with `**strongly**`

File: docs/core-study-details.md [hunk 10]
```diff
@@ -8,98 +8,501 @@ nav_order: 5
...
+|      Column      | Type  |Description|
+|------------------|-------|-----------|
+|cnt               |bigint |count      |
+|cond_category_code|varchar|Encounter Code (Healthcare Setting)|
+|cond_month        |varchar|Month condition recorded|
+|cond_code_display |varchar|Condition code|
 
-## Core study exportable counts tables
+### core__count_documentreference_month
+|     Column      | Type  |Description|
```
Comment (mikix) [inline docs/core-study-details.md:34]: When I was just playing with the previous version of this file, I noticed that the resulting HTML from jekyll that gets put on the doc site required a newline between any section header and the table. Worth testing, but I'd have your generator add a line to be safe.

File: cumulus_library/template_sql/column_datatype.sql.jinja [hunk 11]
```diff
@@ -1,8 +1,13 @@
...
     column_name,
     data_type
 FROM information_schema.columns
 WHERE
     table_schema = '{{ schema_name }}'
-    AND table_name = '{{ table_name|lower }}'
-    AND LOWER(column_name) IN ('{{ column_names|join("', '")|lower }}') --noqa: LT05
+    AND table_name IN ('{{ table_names|join("', '")|lower }}')
+{%- if column_names %}
+AND LOWER(column_name) IN ('{{ column_names|join("', '")|lower }}') --noqa: LT05
```
Comment (mikix) [inline cumulus_library/template_sql/column_datatype.sql.jinja:12]: Why isn't this indented? I don't care, but I thought this repo liked its sensible SQL output.

File: tests/test_data/study_python_valid_generated.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
+## study_python_valid base tables
+
+### study_python_valid__lib_transactions
+|    Column     |  Type   |Description|
```
Comment (mikix) [inline tests/test_data/study_python_valid_generated.md:4]: I might have missed this - how does Description get filled in? Might be nice to have a test row that includes one.

**Review by mikix — APPROVED** [review]

File: cumulus_library/template_sql/column_datatype.sql.jinja (hunk 3 above)
Comment (dogversioning) [reply cumulus_library/template_sql/column_datatype.sql.jinja:4]: i could - i was mirroring the information_schema order here, but that's not the most critical thing to support, I suppose

File: cumulus_library/cli.py (hunk 2 above)
Comment (dogversioning) [reply cumulus_library/cli.py:393]: yeah that worked out well, thanks past collective bikeshedding

File: docs/creating-sql-with-python.md (hunk 5 above)
Comment (dogversioning) [reply docs/creating-sql-with-python.md:99]: nope!

File: cumulus_library/template_sql/column_datatype.sql.jinja (hunk 11 above)
Comment (dogversioning) [reply cumulus_library/template_sql/column_datatype.sql.jinja:12]: I truly cannot figure out why sqlfluff was complaining about this, and end of day brain - the other option is another noqa statement

File: tests/test_data/study_python_valid_generated.md (hunk 12 above)
Comment (dogversioning) [reply tests/test_data/study_python_valid_generated.md:4]: right now - by hand, after the fact. Let me add a note about this someplace.

File: docs/core-study-details.md (hunk 10 above)
Comment (dogversioning) [reply docs/core-study-details.md:34]: oh dear - ok, this might take some work.

Comment (dogversioning) [reply docs/core-study-details.md]: Ok, it's a bit hacky, but that's the default behavior.

File: cumulus_library/template_sql/column_datatype.sql.jinja (hunk 11 above)
Comment (dogversioning) [reply cumulus_library/template_sql/column_datatype.sql.jinja]: noqa it is for this

File: docs/creating-studies.md [hunk 13]
```diff
@@ -95,22 +121,56 @@ Most users have a workflow that looks like this:
...
+  every code for every system found. This is useful when you are specifically
+  looking for data associated with a given clinical coding system, but
+  if you are not careful, you can cause a condition to be counted twice
+  by not specifying a coding system when joining this table with the
+  base condition table.
+
+Your approach to handling this is going to be dictated by the specific
+clinical context you're working with. In cases where we don't specify
+two table types for an array resource, you should assume that we are
+following the second pattern and account for that in your queries.
```
Comment (dogversioning) [inline docs/creating-studies.md:162]: Explicitly calling out this large text block i just added about the denormalized tables.

File: docs/creating-sql-with-python.md [hunk 14]
```diff
@@ -44,6 +48,17 @@ You'll see examples of all three cases in this guide.
...
+If you include a table builder in your study, and you want to see what the
+query being executed looks like, you can use the `generate-sql` command
+in the Cumulus library CLI to write out example queries. They will go into
+a folder inside your study called `reference_sql`.
+
+To document your study strucuture, you can use the `generate-md` command
+to create markdown tables you can copy into your study docs. Note that,
+as of this writing, you'll need to supply a description for each field by
+hand. This output will be generated inside your study, in a file named
+`{study name}_generated.md`.
```
Comment (dogversioning) [inline docs/creating-sql-with-python.md:60]: also calling out the 'you need to define your own description, for now'

**Pull request merged**